dynatrace/configuration_v1/anomaly_detection_process_groups.py

```diff
@@ -0,0 +1,43 @@
+from typing import List, Optional, Dict, Any
+from requests import Response
+
+from dynatrace.environment_v2.schemas import ConfigurationMetadata
+from dynatrace.environment_v2.monitored_entities import EntityShortRepresentation
+from dynatrace.pagination import PaginatedList
+from dynatrace.dynatrace_object import DynatraceObject
+from dynatrace.http_client import HttpClient
+
+class AvailabilityMonitoringPG(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.method: str = raw_element.get("method")
+        self.minimum_threshold: int = raw_element.get("minimumThreshold") 
+
+class AnomalyDetectionPG(DynatraceObject):
+    def _create_from_raw_data(self, raw_element: Dict[str, Any]):
+        self.availability_monitoring: AvailabilityMonitoringPG = AvailabilityMonitoringPG(raw_element=raw_element.get("availabilityMonitoring"))
+
+class AnomalyDetectionPGService:
+    def __init__(self, http_client: HttpClient):
+        self.__http_client = http_client
+
+    def get_configuration(self, id: str):
+        """
+        Get the anomaly detection configuration for the specified process group.
+        """
+        return AnomalyDetectionPG(raw_element=self.__http_client.make_request(f"/api/config/v1/anomalyDetection/processGroups/{id}").json())
+
+    def update_configuration(self, id: str, method: str, minimum_threshold: Optional[int] = None):
```

Comment:
This should receive a `AvailabilityMonitoringPG` instance (or a `AnomalyDetectionPG`, but I see no point as it just wraps around the `AvailabilityMonitoringPG` object). And then call `instance.put()` on the instance that it receives.

If you want an easy way to create an `AvailabilityMonitoringPG` then use a `create_configuration` method that returns this object, which then can be modified and used on the `update` method, or the user could call `instance.put()` directly

Just like this: 

https://github.com/dlopes7/dt-api/blob/69bb84784da6cb7861b9d0eeeb226bcd8c0af143/dynatrace/configuration_v1/extensions.py#L64

https://github.com/dlopes7/dt-api/blob/69bb84784da6cb7861b9d0eeeb226bcd8c0af143/dynatrace/configuration_v1/extensions.py#L137

https://github.com/dlopes7/dt-api/blob/69bb84784da6cb7861b9d0eeeb226bcd8c0af143/dynatrace/configuration_v1/extensions.py#L102

Reply:
I think in the update_configuration method we'd still need to pass a PG id, because unlike the extension configurations it doesn't include anything like an entity_id so that would need to be called.

```
def update_configuration(self, id: str, anomaly_detection_config: AnomalyDetectionPG):
        """
        Update the anomaly detection configuration for the specified process group.
        """
        if anomaly_detection_config.availability_monitoring.method == Method.MINIMUM_THRESHOLD:
            body = {"availabilityMonitoring": {"method": anomaly_detection_config.availability_monitoring.method.value, "minimumThreshold": anomaly_detection_config.availability_monitoring.method.minimum_threshold}}
        else:
            body = {"availabilityMonitoring": {"method": anomaly_detection_config.availability_monitoring.method.value}}
        return self.__http_client.make_request(f"/api/config/v1/anomalyDetection/processGroups/{id}", method="PUT", params=body)
```

Reply:
Implemented this, but since the config doesn't include the entity ID in it at this time, we still must provide the ID in the method call.